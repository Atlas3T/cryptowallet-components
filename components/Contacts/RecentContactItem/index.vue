<template>
  <div class="contact-list-item">
    <div @click.prevent="goToContact">
      <div class="circle" />
      {{ contact.displayName }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'RecentContactItem',
  props: {
    contact: {
      type: Object,
      required: true,
    },
  },
  methods: {
    goToContact() {
      this.$router.push({ path: `/contact-item/${this.contact.id}` });
    },
  },
};
</script>

<style scoped>
.contact-list-item > div {
  padding-right: 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  white-space: nowrap;
}

.circle {
  background: #ececec;
  border-radius: 100%;
  height: 2.5rem;
  width: 2.5rem;
  margin-top: 0.1rem;
  margin-bottom: 0.5rem;
}
</style>
