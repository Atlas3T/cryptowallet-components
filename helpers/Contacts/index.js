import Vue from 'vue';

/**
 * Handles the contacts import and saves them in the store
 */
export default function Contacts() {
  this.vm = new Vue();
  this.import = (contacts) => {
    this.contacts = contacts;
    this.vm.$once('contactsImported', () => { return this.success(); });
    this.vm.$once('contactsImportFailure', () => { return this.failure(); });
    this.vm.$once('contactsImportError', (event) => { return this.error(event.payload); });

    try {
      const options = new ContactFindOptions();
      options.multiple = true;
      const fields = ['*'];
      navigator.contacts.find(fields, (contactsArray) => {
        this.contacts.contacts = contactsArray;
        this.vm.$emit('contactsImported');
      }, () => {
        this.vm.$emit('contactsImportFailure');
      }, options);
    } catch (err) {
      this.vm.$emit('contactsImportError', { payload: err });
    }

    return this;
  };

  this.on = (event, callback) => {
    this[event] = callback;
    return this;
  };
}
