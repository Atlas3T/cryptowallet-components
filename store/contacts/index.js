export default {
  namespaced: true,
  state: {
    contacts: [
      {
        id: 1,
        displayName: 'Terry Tibbs',
        address: 'ncousohfodicwjdclwdnc',
      },
    ],
  },
};
