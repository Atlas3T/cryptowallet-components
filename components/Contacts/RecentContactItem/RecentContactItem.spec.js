import { shallowMount } from '@vue/test-utils';
import RecentContactItem from '@/components/Contacts/RecentContactItem';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

describe('RecentContactItem.vue', () => {
  let storeMocks;
  let wrapper;
  let store;
  let router;

  const propsData = {
    contact: {
      id: 1,
      displayName: 'Fio',
    }
  };

  function wrapperInit (options) {
    return shallowMount(RecentContactItem, options);
  }

  function storeInit (custom) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: `/` });
    wrapper = wrapperInit({ i18n, router, localVue, store: storeMocks.store, propsData });
    store = wrapper.vm.$store;
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a div with .contact-list-item class', () => {
    expect(wrapper.contains('div.contact-list-item')).toBe(true);
  });

  it('renders contacts displayName', () => {
    expect(wrapper.html().includes(propsData.contact.displayName)).toBe(true);
  });

  it('navigates to correct route when clicked on', () => {
    expect(store.state.route.path).toBe('/');
    wrapper.find('.contact-list-item > div').trigger('click');
    expect(store.state.route.path).toBe(`/contact-item/${propsData.contact.id}`);
  });
});
