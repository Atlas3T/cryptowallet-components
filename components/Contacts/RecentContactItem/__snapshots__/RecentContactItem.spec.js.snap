// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentContactItem.vue renders and matches snapshot 1`] = `
<div
  class="contact-list-item"
>
  <div>
    <div
      class="circle"
    />
    
    Fio
  
  </div>
</div>
`;
